diff --git a/lib/JSON/Validator/Schema/OpenAPIv3.pm b/lib/JSON/Validator/Schema/OpenAPIv3.pm
index accfdbe..82713e6 100644
--- a/lib/JSON/Validator/Schema/OpenAPIv3.pm
+++ b/lib/JSON/Validator/Schema/OpenAPIv3.pm
@@ -297,6 +297,8 @@ sub _validate_body {
     # Mutate request content-type if one was not set
     $val->{content_type} //= $negotiated_content_type;
 
+    return if $val->{content_type} =~ m!^multipart/form-data!;
+
     local $self->{coerce}{arrays} = 1
       if $val->{content_type} =~ m!^(application/x-www-form-urlencoded|multipart/form-data)\s*(;|$)!;
 
